src/bpftrace.cpp

```diff
@@ -427,16 +429,21 @@ void perf_event_printer(void *cb_cookie, void *data, int size)
     bpftrace->out_->message(MessageType::cat, buf.str(), false);
 
     return;
-  }
-
-  // printf
-  auto &fmt = std::get<0>(bpftrace->resources.printf_args[printf_id]);
-  auto &args = std::get<1>(bpftrace->resources.printf_args[printf_id]);
-  auto arg_values = bpftrace->get_arg_values(args, arg_data);
+  } else if (printf_id >= AsyncAction::printf &&
+             printf_id <= AsyncAction::printf_end) {
+    auto id = static_cast<size_t>(printf_id) -
+              static_cast<size_t>(AsyncAction::printf);
+    auto &fmt = std::get<0>(bpftrace->resources.printf_args[id]);
+    auto &args = std::get<1>(bpftrace->resources.printf_args[id]);
+    auto arg_values = bpftrace->get_arg_values(args, arg_data);
 
-  bpftrace->out_->message(MessageType::printf,
-                          fmt.format_str(arg_values),
-                          false);
+    bpftrace->out_->message(MessageType::printf,
+                            fmt.format_str(arg_values),
+                            false);
+    return;
+  } else {
+    LOG(WARNING) << "Unknown printf_id: " << static_cast<int64_t>(printf_id);
```

Comment:
`LOG(BUG)` instead? Or at least warn that an event is dropped here?

Reply:
Sure, `LOG(BUG)` works.